frog/imports/client/SingleActivity/TopBar.js

```diff
@@ -1,29 +1,38 @@
 // @flow
 
-import React from 'react';
+import * as React from 'react';
 import { withStyles } from '@material-ui/styles';
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
 import Typography from '@material-ui/core/Typography';
 import Button from '@material-ui/core/Button';
 import { type PropsT } from './types';
 import { style } from './style';
-
+import { withModal } from 'frog-utils';
+import { SignUpModal } from '/imports/client/AccountModal';
 /**
  * Navigation bar displayed at the top
  */
 function TopBar(props: PropsT) {
-  const { classes } = props;
+  const { classes, showModal } = props;
+
+  const openSignUpModal = () => {
+    showModal(<SignUpModal />);
+  };
+
   return (
     <AppBar position="static" color="default">
       <Toolbar classes={{ root: classes.navbar }}>
         <Typography variant="h6" color="inherit" className={classes.logo}>
           FROG
         </Typography>
         <Button size="medium">Help</Button>
-        <Button size="medium">Log In/Sign Up</Button>
+        <Button size="medium" onClick={openSignUpModal}>
+          Log In/Sign Up
+        </Button>
       </Toolbar>
     </AppBar>
   );
 }
-export default withStyles(style)(TopBar);
+const TopBarWithModal = withModal(TopBar);
+export default withStyles(style)(TopBarWithModal);
```

Comment:
NIT: Look at lodash `flow` to compose HOCs: https://lodash.com/docs/4.17.14#flow
``` javascript
// assuming lodash is imported
export default _.flow(withModal, withStyle(style))(TopBar);
```

Reply:
Thanks! Fixed